client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java

```diff
@@ -738,19 +768,22 @@ private SendResult sendDefaultImpl(
 
     private TopicPublishInfo tryToFindTopicPublishInfo(final String topic) {
         TopicPublishInfo topicPublishInfo = this.topicPublishInfoTable.get(topic);
+
+        // topic route info not in local cache, then make a request to nameServer
         if (null == topicPublishInfo || !topicPublishInfo.ok()) {
             this.topicPublishInfoTable.putIfAbsent(topic, new TopicPublishInfo());
```

Comment:
If the topicPublishInfo is not ok(i.e. the topic does not exist), it will generate RPC for each message, which will cause too many requests to the nameserver, and make the nameserver to be hot.

For the failed topic,  we may need to add a placeholder for it.